diff --git a/dist/ts3.9/blocks/DocsContainer.d.ts b/dist/ts3.9/blocks/DocsContainer.d.ts
index be330e44bebb02eaf2c92d365d4e7dc1da452465..6c8b1d42bea2e184456e2757eb2ee20076ba43b3 100644
--- a/dist/ts3.9/blocks/DocsContainer.d.ts
+++ b/dist/ts3.9/blocks/DocsContainer.d.ts
@@ -1,7 +1,8 @@
-import { FunctionComponent } from 'react';
+import { FunctionComponent, ReactNode } from 'react';
 import { AnyFramework } from '@storybook/csf';
 import { DocsContextProps } from './DocsContext';
 export interface DocsContainerProps<TFramework extends AnyFramework = AnyFramework> {
     context: DocsContextProps<TFramework>;
+    children?: ReactNode;
 }
 export declare const DocsContainer: FunctionComponent<DocsContainerProps>;
