diff --git a/dist/builder-manager/index.js b/dist/builder-manager/index.js
index c68536f748afc0397c5522102b239d899cd5cb61..b997503dd116d5241b4781b643d10ee03cc8ed1c 100644
--- a/dist/builder-manager/index.js
+++ b/dist/builder-manager/index.js
@@ -2001,7 +2001,7 @@ var mr = /^\/($|\?)/, L, A, _t = /* @__PURE__ */ o(async (e) => {
       // legacy font formats
       ".woff": "dataurl",
       ".eot": "dataurl",
-      ".ttf": "dataurl"
+      ".ttf": "dataurl",
     },
     target: re,
     supported: st,
@@ -2015,7 +2015,7 @@ var mr = /^\/($|\?)/, L, A, _t = /* @__PURE__ */ o(async (e) => {
     // turn this on to assist with debugging the bundling of managerEntries
     // treeShaking: true,
     sourcemap: !1,
-    conditions: ["browser", "module", "default"],
+    conditions: ["@grafana-app/source", "browser", "module", "default"],
     jsxFactory: "React.createElement",
     jsxFragment: "React.Fragment",
     jsx: "transform",
@@ -2024,17 +2024,17 @@ var mr = /^\/($|\?)/, L, A, _t = /* @__PURE__ */ o(async (e) => {
     legalComments: "external",
     plugins: [Z(pr), (0, At.pnpPlugin)()],
     banner: {
-      js: "try{"
+      js: "try{",
     },
     footer: {
-      js: '}catch(e){ console.error("[Storybook] One of your manager-entries failed: " + import.meta.url, e); }'
+      js: '}catch(e){ console.error("[Storybook] One of your manager-entries failed: " + import.meta.url, e); }',
     },
     define: {
       "process.env": JSON.stringify(a),
       ...cr(a),
       global: "window",
-      module: "{}"
-    }
+      module: "{}",
+    },
   };
 }, "getConfig"), jt = {
   get: /* @__PURE__ */ o(async () => {
