diff --git a/pkg/services/chats/sql_storage.go b/pkg/services/chats/sql_storage.go
index d355ac5770..916dac079c 100644
--- a/pkg/services/chats/sql_storage.go
+++ b/pkg/services/chats/sql_storage.go
@@ -50,10 +50,11 @@ func (s *sqlStorage) CreateMessage(ctx context.Context, orgId int64, ctId int, o
 		if !has {
 			chat.Created = nowUnix
 			chat.Updated = nowUnix
-			chatId, err = dbSession.Insert(&chat)
+			_, err = dbSession.Insert(&chat)
 			if err != nil {
 				return err
 			}
+			chatId = chat.Id
 		}
 		message := Message{
 			ChatId:  chatId,
diff --git a/public/app/plugins/panel/timeseries/plugins/annotations/AnnotationTooltip.tsx b/public/app/plugins/panel/timeseries/plugins/annotations/AnnotationTooltip.tsx
index faebc33721..33a1977572 100644
--- a/public/app/plugins/panel/timeseries/plugins/annotations/AnnotationTooltip.tsx
+++ b/public/app/plugins/panel/timeseries/plugins/annotations/AnnotationTooltip.tsx
@@ -102,7 +102,7 @@ const getStyles = (theme: GrafanaTheme2) => {
     chatWrapper: css`
       margin-top: 10px;
       border-top: 2px solid #2d2b34;
-      max-height: 400px;
+      max-height: 300px;
       overflow-y: scroll;
       padding: 0 3px;
     `,
